### ISSUE TITLE
State monad code doesn't work anymore?

### ISSUE THREAD

**Issue by jamesdbrock** [IHaskell/learn-you-a-haskell-notebook#18]
Check this out.

https://github.com/learnyouahaskell/learnyouahaskell.github.io/issues/7

**Comment by jamesdbrock** [comment]
The `newtype State` code is not a problem because all that code is in raw cells, not code cells.

However there is a problem in Chapter 13, cell 138:

```
instance Monad Prob where
    return x = Prob [(x,1%1)]
    m >>= f = flatten (fmap f m)
    fail _ = Prob []
```

```
<interactive>:4:5: error: ‘fail’ is not a (visible) method of class ‘Monad’
```

Here's why: https://gitlab.haskell.org/haskell/prime/-/wikis/libraries/proposals/monad-fail

**Comment by jamesdbrock** [comment]
Fixed in 53bfb01ca381b28a0d5c15caedbf27e834d05153 de18126b3adae2336154811b15cad1664bbbf4cc
